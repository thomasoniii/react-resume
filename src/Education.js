import React from 'react';

import './styles/Education.css';

export default ({ schools }) => {
  return (
    <div className='section education container-fluid'>
      <div className='row'>
        <div className='col section-header'>Education</div>
      </div>
      {schools.map( school => {
        return (
          [
            <div className='row' key='name'>
              <div className='col school'>
                <span className='school'>{ school.school }</span>,
                <span className='location'> { school.location}</span>
              </div>
              <div className='col'>
                <div className='float-right date'>
                  { school.date }
                </div>
              </div>
            </div>,
            <div className='row' key='results'>
              <div className='col results'>{ school.results}</div>
            </div>
          ]
        )
      })}
    </div>
  )
}
