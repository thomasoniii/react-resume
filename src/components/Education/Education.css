.school-container {
  display: flex;
  justify-content: space-between;
}

.results {
  font-style: italic;
}
