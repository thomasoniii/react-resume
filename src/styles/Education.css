.education, .education ul {
  list-style-type : none
}

.education > li:first-child {
  font-weight : bold;
  font-size : 3vh;
}

.school {
  font-weight : bold;
}

.results {
  font-style : italic;
}
