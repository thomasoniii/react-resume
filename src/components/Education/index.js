import React from "react"
import PropTypes from "prop-types"
import { Card, Typography, List } from "antd"

import "./Education.css"

const { Title, Text } = Typography

const Education = ({ schools }) => (
  <Card size="small" title={<Title level={2}>Education</Title>}>
    <List
      dataSource={schools}
      renderItem={(school) => (
        <List.Item className="school">
          <List.Item.Meta
            title={
              <div className="school-container">
                <span>
                  <span className="school-name">
                    <Text strong>{school.school}</Text>
                  </span>
                  , <span className="location">{school.location}</span>
                </span>
                <span className="date">{school.date}</span>
              </div>
            }
            description={school.results}
          />
        </List.Item>
      )}
    />
  </Card>
)

Education.propTypes = {
  schools: PropTypes.object,
}

export default React.memo(Education)
