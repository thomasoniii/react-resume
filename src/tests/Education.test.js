import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Education from '../components/Education';
import resume from '../data/resume-example.json';

// <Education  filters={ filters } schools={resume.education}     key='education'/>


const filters = {};

describe('Education tests', () => {
  test('Education exists properly', () => {
    const education = renderer.create(
      <Education
        filters={filters}
        schools={resume.education}
      />
    );

    let tree = education.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
