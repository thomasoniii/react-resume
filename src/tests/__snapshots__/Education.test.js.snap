// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education exists properly 1`] = `
<div
  className="section education container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col section-header"
    >
      Education
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm school"
    >
      <span
        className="school"
      >
        University College
      </span>
      ,
      <span
        className="location"
      >
         
        University Town, IL
      </span>
    </div>
    <div
      className="col-sm"
    >
      <div
        className="float-right date"
      >
        June 2012
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col results"
    >
      BS in Ukelele, Minor in Water-walking
    </div>
  </div>
</div>
`;
